src/main/java/net/bramp/ffmpeg/FFmpeg.java

```diff
@@ -166,6 +171,37 @@ private void checkIfFFmpeg() throws IllegalArgumentException, IOException {
     return formats;
   }
 
+  public synchronized List<PixelFormat> pixelFormats() throws IOException {
+    checkIfFFmpeg();
+
+    if (this.pixelFormats == null) {
+      pixelFormats = new ArrayList<>();
+
+      Process p = runFunc.run(ImmutableList.of(path, "-pix_fmts"));
+      try {
+        BufferedReader r = wrapInReader(p);
+        String line;
+        while ((line = r.readLine()) != null) {
+          Matcher m = PIXEL_FORMATS_REGEX.matcher(line);
+          if (!m.matches())
+            continue;
+          String flags = m.group(1);
+
+          pixelFormats.add(new PixelFormat(m.group(2),
+                                           Integer.parseInt(m.group(3)),
+                                           Integer.parseInt(m.group(4)),
+                                           flags));
+        }
+
+        throwOnError(p);
```

Comment:
Do we want to add:
```
this.formats = ImmutableList.copyOf(formats);
```
As that will avoid the caller from editing the list.

Reply:
That would be wise.
Change is implemented

Reply:
Ah sorry... Does this compile? We should be using `pixelFormats` but `formats`. (That was lazy copy and paste on my part).

Reply:
That was a close call.
I noticed it while reading but forgot to change it after copying and pasting.
Yes, it would have compiled but produced a RuntimeException if you didn't call `formats()` beforehand.

Has been changed to `this.pixelFormats = ImmutableList.copyOf(pixelFormats);`

Reply:
Hopefully tests would have caught this :-)... Maybe we should get Travis checking on all PR to make our lives easier.

Reply:
They would have if I'd run them after changing.
Travis on PR would be a nice addition; not sure what the effort for that one is though